### PR TITLE
[124] Withdraw a bid (Frontend)

### PR DESCRIPTION
**Pull request by khasanrashidov**
### Where

[NL-124 Withdraw a bid (Frontend)](https://jjbcapital.atlassian.net/browse/NL-124)

### What

Developing the frontend modal component to enable buyers to withdraw their bids (according to rules in backend).

### How

Implemented the features on frontend fully on buyer lot profile. Now buyers can withdraw their bids. All validation cases are considered and proper messages provided for any success and error cases.

### Affected Area

Auctionify.API.ClientApp
Auctionify.Application.Features.Users

### PR Checklist

_Build_

- [x] Build Succeeded

_Unit Tests_

- [x] All Tests Passed, 0 Ignored, 0 Failed

### CODE REVIEW

**Review by SaidganiUsm — CHANGES_REQUESTED** [review]
Снимок экрана 2023-12-09 211502

Everything is perfectly done. But I will recommend adding a tiny bit of fix as a closing btn in the window of placing the bid.

**Comment by khasanrashidov** [comment]
> Снимок экрана 2023-12-09 211502
> 
> Everything is perfectly done. But I will recommend adding a tiny bit of fix as a closing btn in the window of placing the bid.

There was no close button in figma (and it is another ticket (approved, it is in dev)). But yeah, thank you for recomendation.

**Review by SaidganiUsm — DISMISSED** [review]
Perfect

**Review by SaidganiUsm — APPROVED** [review]
Checked 😀

**Review by otash8110 — DISMISSED** [review]

File: src/Auctionify.API/ClientApp/src/app/components/general/lot-profile/lot-profile.component.ts [hunk 1]
```diff
@@ -176,6 +178,35 @@ export class LotProfileComponent implements OnInit {
         });
     }
 
+    openWithdrawBidModal(): void {
+        this.getBuyerRecentBidForLot().then((bidId) => {
```
Comment (TsvirkoNikita) [inline src/Auctionify.API/ClientApp/src/app/components/general/lot-profile/lot-profile.component.ts:182]: What the reason to get recent bid for user each time when user click on Withdraw button, if you already have bids list in lotData?

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]
Can you only show the "Withdraw" button if the last bid is the current user's bid?

File: src/Auctionify.API/ClientApp/src/app/components/general/lot-profile/lot-profile.component.ts (hunk 1 above)
Comment (khasanrashidov) [reply src/Auctionify.API/ClientApp/src/app/components/general/lot-profile/lot-profile.component.ts:182]: I am taking each user's bid, to check if user can withdraw bid or not. There is a bidRemoved flag in user's bid, so when I get the user's bid, if that flag is true, the user cannot withdraw his bid. As we discussed the buyer can withdraw only once for the lot.

Comment (TsvirkoNikita) [reply src/Auctionify.API/ClientApp/src/app/components/general/lot-profile/lot-profile.component.ts:182]: But you have a bids list inside the lotData object in your component, still don't understand

**Review by TsvirkoNikita — APPROVED** [review]

**Review by TsvirkoNikita — DISMISSED** [review]

**Review by SaidganiUsm — APPROVED** [review]

**Review by otash8110 — APPROVED** [review]

**Review by TsvirkoNikita — APPROVED** [review]

**Pull request merged**